packages/storage/src/platform/node/connection.ts

```diff
@@ -59,16 +59,29 @@ abstract class FetchConnection<T extends ConnectionType>
     this.sent_ = true;
 
     try {
-      const response = await this.fetch_(url, {
+      const response = await this.fetch_({
         method,
-        headers: headers || {},
-        body: body as ArrayBufferView | string
+        url,
+        data: body,
+        headers,
+        responseType: 'arraybuffer'
       });
       this.headers_ = response.headers;
       this.statusCode_ = response.status;
       this.errorCode_ = ErrorCode.NO_ERROR;
-      this.body_ = await response.arrayBuffer();
+      this.body_ = response.data;
     } catch (e) {
+      if (
+        e instanceof AxiosError &&
+        (e as AxiosError)?.response?.status === 404
+      ) {
+        const axiosError = e as AxiosError;
+        this.headers_ = axiosError?.response?.headers;
+        this.statusCode_ = axiosError?.response?.status;
+        this.errorCode_ = ErrorCode.NO_ERROR;
```

Comment:
If there was an error here, why are we setting the `errorCode_` to `ErrorCode.NO_ERROR`?

Reply:
This specific if statement here is needed to match the behavior of node-fetch with axios - In the case of 404 in node-fetch there was no error thrown and the 404 is just preserved in the response data and errorCode is set to NO_ERROR. Axios throws an error on 404 and some tests were breaking when errorCode is not set to NO_ERROR.